### PR TITLE
dc2Deployment code Changes

### PR DESCRIPTION
**Pull request by drajshek**
Added additonal logic for Deployment.Strategy type.

### CODE REVIEW

File: pkg/dc2deployment/dc2deployment.go [hunk 1]
```diff
@@ -96,13 +87,63 @@ func Mutate(pluginName string, dc dcAPI.DeploymentConfig, log *logrus.Logger) (d
...
-	log.Debugf("[%s] d.Selector: %#v", pluginName, deploy.Spec.Selector)
+	log.Debugf("[%s] Strategy: %#v", pluginName, dc.Spec.Strategy)
+
+	// openshift supoorts Strategy.Type={"Recreate","Custom","Rolling"}
+	// K8s native supports k8s.Strategy.Type={"Recreate","RollingUpdate"}
+	// Custom strategy.type is not supported in K8s native, hence defaulting it to RollingUpdate
+	if dc.Spec.Strategy.Type != "" {
+		if dc.Spec.Strategy.Type == "Rolling" || dc.Spec.Strategy.Type == "Custom" {
+			deploy.Spec.Strategy.Type = deployAPI.DeploymentStrategyType("RollingUpdate")
+		} else {
```
Comment (brito-rafa) [inline pkg/dc2deployment/dc2deployment.go:85]: Let's make a comment here that this is expected to be "Recreate"

File: pkg/dc2deployment/dc2deployment.go [hunk 2]
```diff
@@ -96,13 +87,63 @@ func Mutate(pluginName string, dc dcAPI.DeploymentConfig, log *logrus.Logger) (d
...
 
 	if dc.Spec.Template != nil {
 		dc.Spec.Template.DeepCopyInto(&deploy.Spec.Template)
 	}
-
-	log.Tracef("[%s] deployment %#v", pluginName, deploy)
-
+	// End of Spec Section
+
+	//Status Section Start
```
Comment (brito-rafa) [inline pkg/dc2deployment/dc2deployment.go]: We do not need any conversion on the Status field, please drop all the conversion logic. Reason: The Status field is populated by the controller responsible for the object. In this case, we are creating a brand-new object and we have to let the controller handle those. Populating the Status from a different object running on a cluster is misleading.

File: pkg/dc2deployment/dc2deployment.go [hunk 3]
```diff
@@ -96,13 +87,63 @@ func Mutate(pluginName string, dc dcAPI.DeploymentConfig, log *logrus.Logger) (d
...
+			deploy.Status.Conditions[i].Status = dc.Status.Conditions[i].Status
+			deploy.Status.Conditions[i].LastUpdateTime = dc.Status.Conditions[i].LastUpdateTime
+			deploy.Status.Conditions[i].LastTransitionTime = dc.Status.Conditions[i].LastTransitionTime
+			deploy.Status.Conditions[i].Reason = dc.Status.Conditions[i].Reason
+			deploy.Status.Conditions[i].Message = dc.Status.Conditions[i].Message
+		}
+	}
+	// End of Status Section
+
+	//Logging the unsupported fields
```
Comment (brito-rafa) [inline pkg/dc2deployment/dc2deployment.go]: My preference is to annotate on the new object only if the original object had the unsupported field set. There are some not-so-popular deploymentConfig settings that might not make to the annotation of the httpproxy. Having said that, please add conditions (when it is fit) to check if the field exists before creating the annotation.

**Review by brito-rafa — CHANGES_REQUESTED** [review]
I added some comments - there are a couple of changes. I love the new strategy condition - much more robust. We do not need any Status field condition and I want to be more selective on the annotation.

File: pkg/dc2deployment/dc2deployment.go (hunk 1 above)
Comment (drajshek) [reply pkg/dc2deployment/dc2deployment.go:85]: Added comment for the Recreate Strategy.Type.

File: pkg/dc2deployment/dc2deployment.go (hunk 2 above)
Comment (drajshek) [reply pkg/dc2deployment/dc2deployment.go]: Removed the status filed assignment.

File: pkg/dc2deployment/dc2deployment.go (hunk 3 above)
Comment (drajshek) [reply pkg/dc2deployment/dc2deployment.go]: Removed unwanted annotations, retaining only test, trigger,and activeDeadlineSeconds.

**Review by brito-rafa — APPROVED** [review]

**Pull request merged**